test/blackbox-tests/test-cases/melange/runtime-deps.t

```diff
@@ -0,0 +1,97 @@
+Test runtime_deps field
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.7)
+  > (using melange 0.1)
+  > EOF
+
+  $ cat > dune <<EOF
+  > (melange.emit
+  >  (target output)
+  >  (entries main)
+  >  (libraries lib)
+  >  (module_system commonjs))
+  > 
+  > (library
+  >  (name lib)
+  >  (modules lib)
+  >  (modes melange)
+  >  (melange.runtime_deps assets/file.txt))
+  > EOF
+
+  $ cat > lib.ml <<EOF
+  > let dirname = [%bs.raw "__dirname"]
+  > let file_path = "assets/file.txt"
+  > let file_content = Node.Fs.readFileSync (dirname ^ "/" ^ file_path) \`utf8
+  > let print_file () = Js.log file_content
+  > EOF
+
+  $ mkdir assets
+  $ cat > assets/file.txt <<EOF
+  > hello from file
+  > EOF
+
+  $ cat > main.ml <<EOF
+  > let () = Lib.print_file ()
+  > EOF
+
+  $ dune rules output/main.js
+  Error: Dependency cycle between:
+     Computing directory contents of _build/default
+  -> Evaluating predicate in directory _build/default/output/assets
+  -> Computing directory contents of _build/default
+  [1]
```

Comment:
@rgrinberg do you have any ideas how to fix this dep cycle? It seems it is problematic to use globs with melange runtime_deps, should I just skip glob functionality for now?

Reply:
I managed to get over this error by removing glob functionality in https://github.com/ocaml/dune/pull/6643/commits/d330036a741125d87d5dfc034e66b97cecf686a4. It is less convenient to not have globs, but if it is going to lead to confusing rule cycles errors, better to leave them out.

However, runtime_deps are still not copied, I believe because the copy rules are there, but nothing depend on the asset files. I wonder though, what exactly should depend on these files? Should these asset paths be passed down and added as dep of the `cmj` creation rules for each module? Otherwise I am not sure where else they could be added, as there is no such thing as "library-level target" in melange, like `cma`.

Reply:
The targets should be added to the alias in the melange.emit stanza I'm thinking

You can use globs as long as you factor things into directories properly. You can also support globs and explicit files with constructors. E.g. `(foo.txt (glob_files %{project_root}/*.txt))`

Reply:
> The targets should be added to the alias in the melange.emit stanza I'm thinking

Is there a possibility to group all these copy rules under some "internal" alias that `melange.emit` can use? Duplicating the targets in melange.emit stanzas kind of defeats the point of having `runtime_deps` field, as it would roughly be the same as using `copy_files`.

Reply:
One more thought about this coupling between library assets and emit stanza. In a future, there will be authors that publish Melange libraries. These libraries might need some assets to work properly. The problem as we are discussing is that the copying rules + alias deps are only known at melange.emit stanza time. But this is outside the control of the library author.

I am not sure how to represent this in Dune, but it seems we need some kind of "lazy rule", that contains all the information about the assets copying rules, except the prefix path which is the emit target. Then, at emission time, the rules would be "applied" with this prefix. 

cc @anmonteiro

Reply:
Another option is to "standardize" the output folders as we have discussed at some point in the past, always choosing either `es6` or `commonjs` target folders, depending on `module_system`. Or just remove `target`, as was implemented by Antonio in https://github.com/ocaml/dune/pull/6541/.